operator/v1/0000_50_cluster_storage_operator_01_crd.yaml

```diff
@@ -69,6 +78,9 @@ spec:
                   type: object
                   nullable: true
                   x-kubernetes-preserve-unknown-fields: true
+              x-kubernetes-validations:
```

Comment:
We should add integration tests for validations into https://github.com/openshift/api/blob/master/operator/v1/stable.storage.testsuite.yaml

The documentation for how the tests work is https://github.com/openshift/api/tree/master/tests#readme

For these, I would expect to see some "onupdate" style tests, so, something like:

```
onUpdate:
- name: Should allow enabling the vSphere migration for vSphere
  initial: |
    apiVersion: operator.openshift.io/v1
    kind: Storage
    spec: {} # No spec is required for a Storage
   updated: |
     apiVersion: operator.openshift.io/v1
     kind: Storage
     spec:
       csiMigrationVSphere: Enabled
   expected: |
    apiVersion: operator.openshift.io/v1
    kind: Storage
    spec:
       csiMigrationVSphere: Enabled
       logLevel: Normal
       operatorLogLevel: Normal
- name: Should not allow disabling the vSphere migration for vSphere
  initial: |
    apiVersion: operator.openshift.io/v1
    kind: Storage
    spec:
      csiMigrationVSphere: Enabled
   updated: |
     apiVersion: operator.openshift.io/v1
     kind: Storage
     spec:
       csiMigrationVSphere:""
   expectedError: "csiMigrationVSphere may not be unset once Enabled"
```

You probably want another case for trying to unset the field instead of changing to empty string as well, I'll leave that and any others to you

Reply:
Done, thanks for the references.